bot/persistence/database_connector.py

```diff
@@ -438,40 +437,36 @@ def add_group_offer_and_requests(self, user_id: str,
                 db_manager.execute(queries.INSERT_GROUP_REQUEST, (user_id, course, group_nr))
             db_manager.commit()
 
-    def update_group_exchange_message_id(self, user_id: str,
-                                         course: str,
-                                         message_id: str):
+    def update_group_exchange_message_id(self, user_id: int, course: str, message_id: int):
         """Updates the message id in the GroupOffer table from 'undefined' to a valid value
 
         This function is necessary because the message_id can only be retrieved after the embed is sent, which happens
         after inserting in the db, to ensure constraints are fulfilled.
 
         Args:
-            user_id (str): The user_id of the requesting user.
+            user_id (int): The id of the requesting user.
             course (str): The course that should be exchanged.
-            message_id (str): The id of the message that contains the group exchange embed.
+            message_id (int): The id of the message that contains the group exchange embed.
         """
         with DatabaseManager(self._db_file) as db_manager:
             db_manager.execute(queries.UPDATE_GROUP_MESSAGE_ID, (message_id, user_id, course))
             db_manager.commit()
 
-    def get_candidates_for_group_exchange(self, author_id,
-                                          course: str,
-                                          offered_group: int,
-                                          requested_groups: Iterable[int]):
+    def get_candidates_for_group_exchange(self, user_id: int, course: str, offered_group: int,
+                                          requested_groups: Iterable[int]) -> Optional[List[tuple]]:
         """Gets all possible candidates for a group exchange offer.
 
         Args:
-            author_id (str): The id of the author of the request.
+            user_id (int): The id of the user who created the request.
             course (str): The course for which the candidates are searched.
             offered_group (int): The group that the user offers.
             requested_groups (Iterable[int]): The groups that the user requests.
 
         Returns:
-            (Tuple[str, str]): UserId and MessageId of potential group exchange candidates.
+            Optional[List[tuple]]: A list containing user id and mesage id of potential group exchange candidates.
```

Comment:
Why use a list for a fixed size response? That's what tuples are for.

Reply:
I misread that. Will fix.

Reply:
Fixed size response? As far as I know you get here the potential members with which you could trade. So how is this a fixed size response? There could be 0, 3 or x people to trade with.

Reply:
Nvm, i missread your datatype. I thought you wrote Optional[List[str,str]], so you replaced the tuple with List, but you wrapped it.